hil-test/tests/aes.rs

```diff
@@ -452,6 +456,45 @@ mod tests {
         hil_test::assert_eq!(output, CIPHERTEXT_ECB_128);
     }
 
+    #[test]
+    async fn test_aes_work_queue_work_posted_before_queue_started_async() {
```

Comment:
This was failing before. I still suspect that if an async poll completes a different work item, it may end up never completing itself, more tests are needed.